tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -215,100 +226,378 @@ func (p *PortTestSuite) TestClusterIP() {
+	})
 	require.NoError(p.T(), err)
 
-	hostPort := getHostPort()
+	port := getHostPort()
 	nodePort := getNodePort()
 
 	serviceName := namegen.AppendRandomString("test-service")
 	p.T().Logf("Creating service with name [%v]", serviceName)
 	ports := []corev1.ServicePort{
 		{
 			Protocol:   corev1.ProtocolTCP,
-			Port:       int32(hostPort),
+			Port:       int32(port),
 			TargetPort: intstr.FromInt(defaultPort),
 			NodePort:   int32(nodePort),
 		},
 	}
-	lbService := services.NewServiceTemplate(serviceName, p.namespace.Name, corev1.ServiceTypeLoadBalancer, ports, daemonsetTemplate.Spec.Selector.MatchLabels)
-	serviceResp, err := services.CreateService(steveClient, lbService)
+
+	clusterIPService := services.NewServiceTemplate(serviceName, namespace.Name, corev1.ServiceTypeLoadBalancer, ports, deploymentTemplate.Spec.Template.Labels)
+	serviceResp, err := services.CreateService(steveClient, clusterIPService)
 	require.NoError(p.T(), err)
 
 	err = services.VerifyService(steveClient, serviceResp)
 	require.NoError(p.T(), err)
 
+	log.Info("Scaling up deployment")
+	replicas = int32(3)
+	deploymentTemplate.Spec.Replicas = &replicas
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 3, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+
+	log.Info("Scaling down deployment")
+	replicas = int32(2)
+	deploymentTemplate.Spec.Replicas = &replicas
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 2, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+
+	log.Info("Upgrating deployment")
+	for _, c := range deploymentTemplate.Spec.Template.Spec.Containers {
+		c.Name = namegen.AppendRandomString("test-upgrade")
+	}
+
+	log.Info("Updating deployment replicas")
+	deploymentTemplate, err = deployment.UpdateDeployment(p.client, p.cluster.ID, namespace.Name, deploymentTemplate, true)
+	require.NoError(p.T(), err)
+	p.validateWorkload(deploymentTemplate, containerImage, 2, namespace.Name)
+	p.validateLoadBalance(steveClient, nodePort, deploymentName)
+}
+
+// This must be a valid port number, 10250 < hostPort < 65536
+// The range 1-10250 are reserved to Rancher
+// Using a random port to avoid 'port in use' failures and allow the test to be rerun
+func getHostPort() int {
+	return rand.IntN(55283) + 10251
+}
+
+// It will allocate a port from a range 30000-32767
+// Using a random port to avoid 'port in use' failures and allow the test to be rerun
+func getNodePort() int {
+	return rand.IntN(2767) + 30000
+}
+
+func (p *PortTestSuite) validateWorkload(deployment *appv1.Deployment, image string, expectedReplicas int, namespaceName string) {
```

Comment:
nit; still would ask that these helpers get pulled out into a separate file.

Reply:
This refactor might take some time, so I'll add it to the ticket https://github.com/rancher/qa-tasks/issues/1607 to be addressed in the future, and I'll include this point in the review checklist